src/hotspot/share/opto/escape.cpp

```diff
@@ -1264,7 +1282,24 @@ bool ConnectionGraph::reduce_phi_on_safepoints_helper(Node* ophi, Node* cast, No
   return true;
 }
 
-void ConnectionGraph::reduce_phi(PhiNode* ophi, GrowableArray<Node *>  &alloc_worklist, GrowableArray<Node *>  &memnode_worklist) {
+void ConnectionGraph::reduce_phi(PhiNode* ophi, GrowableArray<Node *>  &alloc_worklist, GrowableArray<Node *>  &memnode_worklist, Unique_Node_List &reducible_merges) {
+  Unique_Node_List nested_phis;
+  // Collect nested phi nodes
+  for (DUIterator_Fast imax, i = ophi->fast_outs(imax); i < imax; i++) {
+    Node* use = ophi->fast_out(i);
+    if (use->is_Phi() && use->_idx != ophi->_idx)  {
```

Comment:
I think you can remove the second part of this condition since you checked it in "can_reduce..." ?! Perhaps just make it an assert.